src/library/trackcollection.cpp

```diff
@@ -143,6 +143,38 @@ void TrackCollection::relocateDirectory(QString oldDir, QString newDir) {
 }
 
 bool TrackCollection::hideTracks(const QList<TrackId>& trackIds) {
+    // Warn if tracks have a playlist membership
+    QSet<int> allPlaylisIds;
+    for (const auto& trackId: trackIds) {
+        QSet<int> playlisIds;
+        m_playlistDao.getPlaylistsTrackIsIn(trackId, &playlisIds);
+        QString playlistNames;
+        for (const auto& playlisId: playlisIds) {
+            if (m_playlistDao.getHiddenType(playlisId) != PlaylistDAO::PLHT_SET_LOG) {
+                allPlaylisIds.insert(playlisId);
+            }
+        }
+    }
+
+    if (allPlaylisIds.count()) {
+        QString playlistNames;
+        for (const auto& playlisId: allPlaylisIds) {
+            playlistNames += "\"" + m_playlistDao.getPlaylistName(playlisId) + "\"\n";
+        }
+
+        if (QMessageBox::question(
+                nullptr,
+                tr("Hiding tracks"),
+                tr("The selected tracks are member of the following playlists:\n\n"
```

Comment:
A little clearer:

The selected tracks are in the following playlists: %1. 
Hiding them from the library will remove them from these playlists. Continue?

It would also be nice to have the newlines out of hte translation string. Maybe they could be part of the %1.